eng/build.ps1

```diff
@@ -148,6 +148,9 @@ if ($vs) {
     # Respect the RuntimeConfiguration variable for building inside VS with different runtime configurations
     $env:RUNTIMECONFIGURATION=$runtimeConfiguration
   }
+  
+  # Restore the solution to workaround https://github.com/dotnet/runtime/issues/32205
+  Invoke-Expression "& (Split-Path $PSScriptRoot -Parent | Join-Path -ChildPath dotnet.cmd) restore $vs"
```

Comment:
You could technically just use `DOTNET_ROOT` or call `dotnet` since we add it to the path for this process.